backend/static.go

```diff
@@ -41,18 +48,62 @@ func handleNotFound(h http.Handler) http.HandlerFunc {
 	})
 }
 
-func serveStaticFiles(h http.Handler) http.Handler {
-	return EnableCors(func(w http.ResponseWriter, r *http.Request) {
-		if r.Host == OLD_ADDRESS || IsInsecureRequest(r) {
-			RedirectToSecureVersion(w, r)
-			return
-		}
-		if strings.HasSuffix(r.URL.Path, ".json") {
-			w.Header().Set("Content-Type", "application/json")
+func proxyGAE(prefix string, r *http.Request) *httputil.ReverseProxy {
+	u, err := url.Parse(prefix)
+	if err != nil {
+		log.Fatal(err)
+	}
+	p := httputil.NewSingleHostReverseProxy(u)
+	// Fiddling with the transport might not be necessary with the Go 1.11 runtime:
+	// https://cloud.google.com/blog/products/application-development/go-1-11-is-now-available-on-app-engine
+	p.Transport = &urlfetch.Transport{
+		Context: appengine.NewContext(r),
+	}
+	return p
+}
+
+func shouldProxy(r *http.Request) bool {
+	act := r.Header.Get("amp-cache-transform")
+	if act == "" {
+		return false
+	}
+	accept := r.Header.Get("accept")
+	if accept != "" {
+		a := strings.Split(accept, ",")
+		for _, v := range a {
+			if strings.HasPrefix(strings.TrimSpace(v), "application/signed-exchange") {
+				return true
+			}
 		}
-		SetDefaultMaxAge(w)
-		h.ServeHTTP(w, r)
-	})
+	}
+	return false
+}
+
+func isAMP(r *http.Request) bool {
```

Comment:
You could build a map of valid AMP files during compilation and use it here.

Reply:
You mean gulp's `build:packager` task could create a `sitemap.json`, which is consulted here to figure out what resources the packager can handle? I can do this if necessary, just adds time + deployment complexity. (Note I'm assuming the proxying to packager never needs to be fast, and that a "slow" 404 if `isAMP()` is wrong is okay.)

Reply:
Just realised that you're doing this already in `build:packager`, should be easy to integrate there. But this is up to you.